modules/iam/groups/locals.tf

```diff
@@ -0,0 +1,129 @@
+locals {
```

Comment:
I kinda like you defined these inside `locals` in contrast to infra repo where they also put it inside `data`. Don't think any conventions for other files such as `data` is there.

One thing to note, even Easyhotel have done is different - they included them in `data` instead  _but_ defined it using the data block syntax instead https://github.com/nimblehq/easyhotel-infra-aws/blob/develop/modules/iam_groups/data.tf. 

Me personally data blocks seems easier to read but I guess both are fine :)


```
allow_manage_own_credentials = jsonencode({
    Version = "2012-10-17"
    Statement = [
      {
        Sid    = "AllowViewAccountInfo"
        Effect = "Allow"
        Action = [
          "iam:GetAccountPasswordPolicy",
          "iam:ListVirtualMFADevices",
          "iam:GetLoginProfile"
        ]
        Resource = "*"
      },
```

vs

```
data "aws_iam_policy_document" "allow_manage_own_access_key" {
  version = "2012-10-17"

  statement {
    sid       = "ManageOwnAccessKeys"
    effect    = "Allow"
    resources = ["arn:aws:iam::*:user/$${aws:username}"]

    actions = [
      "iam:CreateAccessKey",
      "iam:DeleteAccessKey",
      "iam:GetAccessKeyLastUsed",
      "iam:GetUser",
      "iam:ListAccessKeys",
      "iam:UpdateAccessKey"
    ]
  }
```

Reply:
From my reference, `data` stores external retrievable data while `locals` is used to define variables within the Terraform configuration.
In this case, I believe the `locals` is more proper 🤔